src/charm.py

```diff
@@ -527,6 +526,11 @@ def _on_mongod_pebble_ready(self, event) -> None:
             event.defer()
             return
 
+        if any(not storage for storage in self.model.storages.values()):
```

Comment:
four things

1. excellent work digging and getting to the bottom of this
2. noob question, why do we need to check `not storage`
3. please add a comment on why this is needed so it will be obvious in the future
4. Also I think other Data Platform engineers will benefit from this knowledge! Please share if you haven't yet